modules/apps/journal/journal-web/src/test/java/com/liferay/journal/web/internal/portlet/action/util/DataDefinitionFieldNameUtilTest.java

```diff
@@ -0,0 +1,162 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.journal.web.internal.portlet.action.util;
+
+import com.liferay.data.engine.rest.dto.v2_0.DataDefinition;
+import com.liferay.data.engine.rest.dto.v2_0.DataDefinitionField;
+import com.liferay.data.engine.rest.dto.v2_0.DataLayout;
+import com.liferay.data.engine.rest.dto.v2_0.DataLayoutColumn;
+import com.liferay.data.engine.rest.dto.v2_0.DataLayoutPage;
+import com.liferay.data.engine.rest.dto.v2_0.DataLayoutRow;
+import com.liferay.portal.kernel.util.HashMapBuilder;
+import com.liferay.portal.test.rule.LiferayUnitTestRule;
+
+import java.util.Map;
+
+import org.junit.Assert;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+
+/**
+ * @author Carolina Barbosa
+ */
+public class DataDefinitionFieldNameUtilTest {
+
+	@ClassRule
+	@Rule
+	public static final LiferayUnitTestRule liferayUnitTestRule =
+		LiferayUnitTestRule.INSTANCE;
+
+	@Test
+	public void testNormalizeFieldName() {
+		DataDefinition dataDefinition = new DataDefinition() {
+			{
+				dataDefinitionFields = new DataDefinitionField[] {
+					new DataDefinitionField() {
+						{
+							customProperties =
+								HashMapBuilder.<String, Object>put(
+									"fieldReference", "textFieldReference"
+								).build();
+							name = "textFieldName";
+						}
+					},
+					new DataDefinitionField() {
+						{
+							customProperties =
+								HashMapBuilder.<String, Object>put(
+									"fieldReference", "selectFieldReference"
+								).put(
+									"options",
+									HashMapBuilder.put(
+										"en_US",
```

Comment:
Curiosity: Why did you not put the "HashMapBuilder. **<String, Object>** put()" here?

Reply:
I had to put it above in `customProperties = HashMapBuilder.<String, Object>` because the `customProperties` is explicitly expecting a `Map<String, Object>`..  I didn't need to put it here because if the `key` and the `value` have a specific type (in this case, `String` and `Object[]`), the `HashMapBuilder` infers the type for the `Map`.